### ISSUE TITLE
Update tsconfig-paths dependency

### ISSUE THREAD

**Issue by monkollek** [import-js/eslint-plugin-import#2711]
The current version of eslint-plugin-import uses tsconfig-paths 3.14.1

The issue with this version is that it's dependency json5 1.0.1 is dependent on minimist 1.2.0 which has a vulnerability
Minimist <=1.2.5 is vulnerable to Prototype Pollution via file index.js, function setKey() (lines 69-95).

The latest version of tsconfig-paths 4.1.2 uses json 2.2.2 which does not have the minimist 1.2.0 dependency/vulnerability.

**Comment by ljharb** [comment]
1. the vulnerability is in json5, not tsconfig-paths
1. we depend on tsconfig-paths v3, which depends on json5 v1
1. json5 v1 and v2 *both* have the fix published, so all you need to do is update your lockfile
1. this is not actually a valid vulnerability for this plugin anyways, so it's irrelevant
1. tsconfig-paths v4 is *absolutely* a breaking change because it drops, among others, node 4 - see #2447 

This is a duplicate of #2625; a duplicate of #2628; a duplicate of #2626; a duplicate of #2627; a duplicate of #2631; a duplicate of #2632; a duplicate of #2634; a duplicate of #2635; a duplicate of #2636; a duplicate of #2637; a duplicate of #2639; a duplicate of #2642; a duplicate of #2643; a duplicate of #2649; a duplicate of #2655; a duplicate of #2660; a duplicate of #2670; a duplicate of #2692.
